components/bounty/funding-section/controller.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+
+import BigNumber from "bignumber.js";
+import { useTranslation } from "next-i18next";
+
+import { useAppState } from "contexts/app-state";
+
+import { getIssueState } from "helpers/handleTypeIssue";
+
+import { fundingBenefactor } from "interfaces/issue-data";
+
+import FundingSectionView from "./view";
+
+export default function FundingSectionController() {
+  const { t } = useTranslation(["common", "funding"]);
+
+  const [walletFunds, setWalletFunds] = useState<fundingBenefactor[]>();
+
+  const { state } = useAppState();
+
+  const isConnected = !!state.currentUser?.walletAddress;
+  const hasReward = state.currentBounty?.data?.hasReward;
+  const isBountyClosed = !!state.currentBounty?.data?.isClosed;
+  const isBountyFunded = !!state.currentBounty?.data?.isFunded;
+  const isBountyInDraft = !!state.currentBounty?.data?.isDraft;
+  const transactionalSymbol =
+    state.currentBounty?.data?.transactionalToken?.symbol;
+  const rewardTokenSymbol = state.currentBounty?.data?.rewardToken?.symbol;
+
+  const fundsGiven =
+    walletFunds?.reduce((acc, fund) => fund.amount.plus(acc), BigNumber(0)) ||
+    BigNumber(0);
+
+  const futureRewards = fundsGiven
+    .multipliedBy(state.currentBounty?.data?.rewardAmount)
+    .dividedBy(state.currentBounty?.data?.fundingAmount)
+    .toFixed();
+
+  const collapseAction = isBountyClosed
+    ? t("funding:rewards")
+    : t("funding:actions.manage-funding");
+
+  const isCanceled =
+    getIssueState({
+      state: state.currentBounty?.data?.state,
+      amount: state.currentBounty?.data?.amount,
+      fundingAmount: state.currentBounty?.data?.fundingAmount,
+    }) === "canceled";
+
+  useEffect(() => {
+    if (!state.currentUser?.walletAddress || !state.currentBounty?.data) return;
```

Comment:
again, if we use the server-side prop then this won't be needed as funds, and all the needed information can be inferred from the data brought over by parent page-controller